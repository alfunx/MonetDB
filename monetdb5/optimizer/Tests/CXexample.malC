# The 4000 series deals with strategic optimizer constructs
# common subexpression evaluation
function foo();
b:= bat.new(:oid,:int);
c:= bat.new(:oid,:int);       
d:= algebra.subselect(b,0,100,true,true,false);
e:= algebra.subselect(b,0,100,true,true,false);    
k1:= 24;
k2:= 27;
l:= k1+k2;
l2:= k1+k2;             
l3:= l2+k1;             
end;

mdb.List("user","foo");
optimizer.commonTerms("user","foo");
optimizer.aliases("user","foo");
mdb.List("user","foo");

