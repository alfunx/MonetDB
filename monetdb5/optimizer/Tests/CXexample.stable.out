stdout of test 'CXexample` in directory 'monetdb5/optimizer` itself:


# 23:53:03 >  
# 23:53:03 >  Mtimeout -timeout 60 Mserver "--config=/ufs/mk/monet5/src/mal/Tests/All.conf" --debug=10 --set "monet_mod_path=/ufs/mk/monet5/Linux/lib/MonetDB5:/ufs/mk/opensource/MonetDB/Linux/lib/MonetDB" --set "gdk_dbfarm=/ufs/mk/opensource/MonetDB/Linux/var/MonetDB5/dbfarm" --set "sql_logdir=/ufs/mk/opensource/MonetDB/Linux/var/MonetDB5/log" --set mapi_port=39352 --set sql_port=45171 --set xquery_port=52900 --set monet_prompt= --trace --dbname=mTests_src_optimizer  CXexample.mal </dev/null
# 23:53:03 >  

# MonetDB server v5.9.0, based on kernel v1.27.0
# Serving database 'mTests_src_optimizer', using 4 threads
# Compiled for x86_64-unknown-linux-gnu/64bit with 64bit OIDs dynamically linked
# Copyright (c) 1993-July 2008 CWI.
# Copyright (c) August 2008-2015 MonetDB B.V., all rights reserved
# Visit http://www.monetdb.org/ for further information
# Listening for connection requests on mapi:monetdb://vienna.da.cwi.nl:37175/
# Listening for UNIX domain connection requests on mapi:monetdb:///var/tmp/mtest-27236/.s.monetdb.37175
# MonetDB/GIS module loaded
# Start processing logs sql/sql_logs version 52200
# Start reading the write-ahead log 'sql_logs/sql/log.4'
# Finished reading the write-ahead log 'sql_logs/sql/log.4'
# Finished processing logs sql/sql_logs
# MonetDB/SQL module loaded
# MonetDB/R   module loaded

Ready.

# 15:36:19 >  
# 15:36:19 >  "mclient" "-lmal" "-ftest" "-Eutf-8" "-i" "-e" "--host=/var/tmp/mtest-27236" "--port=37175"
# 15:36:19 >  

function user.foo():void;               	#[0] (0)  0 
    b:bat[:int] := bat.new(:oid,:int);  	#[1] (0) CMDBATnew 1 <- 2 3 
    c:bat[:int] := bat.new(:oid,:int);  	#[2] (0) CMDBATnew 4 <- 2 3 
    d:bat[:oid] := algebra.subselect(b:bat[:int],0:int,100:int,true:bit,true:bit,false:bit);	#[3] (0) ALGsubselect1 5 <- 1 6 7 8 9 10 
    l:int := calc.+(24:int,27:int);     	#[4] (0) CMDvarADDsignal 16 <- 13 15 
    l3:int := calc.+(l:int,24:int);     	#[5] (0) CMDvarADDsignal 18 <- 16 13 
end user.foo;                           	#[6] (0)  
function user.foo():void;               	#[0] (0)  0 
    b:bat[:int] := bat.new(:oid,:int);  	#[1] (0) CMDBATnew 1 <- 2 3 
    c:bat[:int] := bat.new(:oid,:int);  	#[2] (0) CMDBATnew 4 <- 2 3 
    d:bat[:oid] := algebra.subselect(b:bat[:int],0:int,100:int,true:bit,true:bit,false:bit);	#[3] (0) ALGsubselect1 5 <- 1 6 7 8 9 10 
    l:int := calc.+(24:int,27:int);     	#[4] (0) CMDvarADDsignal 16 <- 13 15 
    l3:int := calc.+(l:int,24:int);     	#[5] (0) CMDvarADDsignal 18 <- 16 13 
end user.foo;                           	#[6] (0)  

# 15:36:19 >  
# 15:36:19 >  "Done."
# 15:36:19 >  

